src/inmanta/ast/type.py

```diff
@@ -280,16 +386,22 @@ def type_string(self) -> str:
     def type_string_internal(self) -> str:
         return self.type_string()
 
+    def as_python_type_string(self) -> "str | None":
+        return "float"
+
+    def to_python(self, instance: object) -> "object":
+        return instance
+
 
 @stable_api
-class Integer(Number):
+class Integer(Primitive):
```

Comment:
This was irregular (Integrer is a number, float is not)

I updated it, but perhaps I shouldn't?

@sanderr wdyt?

Reply:
I think I was vaguely aware of this, but never dared touch it. I think it will be fine, and I think we have sufficient test coverage now that we'll notice if it isn't.